packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -75,14 +75,6 @@ function validateSlots(vm: VM, html: Template) {
     const { slots = EmptyArray } = html;
 
     for (const slotName in cmpSlots) {
-        // eslint-disable-next-line lwc-internal/no-production-assert
-        assert.isTrue(
-            isArray(cmpSlots[slotName]),
-            `Slots can only be set to an array, instead received ${toString(
-                cmpSlots[slotName]
-            )} for slot "${slotName}" in ${vm}.`
-        );
-
         if (slotName !== '' && ArrayIndexOf.call(slots, slotName) === -1) {
             // TODO [#1297]: this should never really happen because the compiler should always validate
```

Comment:
I would get rid of this comment, I don't think it's possible to add the second pass compilation at this point.